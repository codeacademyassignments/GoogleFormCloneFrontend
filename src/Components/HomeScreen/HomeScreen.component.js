import React, { Component } from 'react';
import {
  View, Text, Alert, TouchableOpacity, ScrollView,
} from 'react-native';
import PropTypes from 'prop-types';
import style from './HomeScreen.style';
import FormBlock from '../FormBlock/FormBlock.component';
import getAllForms from '../../Helper/getAllForms';

export default class HomeScreen extends Component {
  static navigationOptions = () => ({
    headerTitle: (
      <Text style={style.title}>Awesome Forms</Text>
    ),
    headerRight: (
      <View style={style.menuIcon}>
        <View borderRadius={10} style={style.iconCircleEdge} />
        <View borderRadius={10} style={style.iconCircleMiddle} />
        <View borderRadius={10} style={style.iconCircleEdge} />
      </View>
    ),
    headerStyle: { backgroundColor: '#3F50B8', height: 80 },
    headerTitleStyle: { color: 'white' },
  }
  )

  state = {
    forms: [],
  }

  async componentDidMount() {
    const forms = await getAllForms();
    this.setState({ forms: forms.data });
  }

  getFormBlocks = formBlocks => formBlocks.map(formBlock => (
    <FormBlock key={formBlock.formName} formBlock={formBlock} />
  ))

  render() {
    const { forms } = this.state;
    const onPress = () => () => {
      const { navigation } = this.props;
      navigation.navigate('CreateForm');
    };
    return (
      <ScrollView contentContainerStyle={{ height: '100%' }}>
        <View style={style.formBlocks}>
          {this.getFormBlocks(forms)}
          <View style={{ alignSelf: 'flex-end', margin: 10 }}>
            <TouchableOpacity
              onPress={onPress()}
              style={style.createFormButton}
            >
              <View style={style.cross}>
                <View style={style.crossUp} />
                <View style={style.crossFlat} />
              </View>
            </TouchableOpacity>
          </View>
        </View>
      </ScrollView>

    );
  }
}

HomeScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};
