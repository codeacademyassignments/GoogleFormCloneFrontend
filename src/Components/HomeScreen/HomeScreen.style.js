import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  menuIcon: {
    flex: 1,
    justifyContent: 'center',
    marginRight: 10,
  },
  title: {
    fontSize: 22,
    color: 'white',
    marginLeft: 15,
  },
  iconCircleMiddle: {
    backgroundColor: 'white',
    width: 5,
    height: 5,
    marginTop: 5,
    marginBottom: 5,
  },
  iconCircleEdge: {
    backgroundColor: 'white',
    width: 5,
    height: 5,
  },
  formBlocks: {
    // backgroundColor: '#D6E4F1',
    alignItems: 'center',
  },
  createFormButton: {
    borderWidth: 1,
    borderColor: 'rgba(0,0,0,0.2)',
    alignItems: 'center',
    justifyContent: 'center',
    width: 70,
    bottom: 0,
    right: 45,
    height: 70,
    backgroundColor: '#3F50B8',
    borderRadius: 100,
  },
  forms: {
    flex: 5,
    backgroundColor: '#D6E4F1',
  },
  crossUp: {
    backgroundColor: 'white',
    height: 20,
    width: 5,
  },
  crossFlat: {
    backgroundColor: 'white',
    height: 5,
    width: 20,
    position: 'absolute',
    left: -8,
    top: 8,
  },
});
