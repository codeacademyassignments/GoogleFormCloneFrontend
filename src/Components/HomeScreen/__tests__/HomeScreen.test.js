import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import HomeScreen from '../HomeScreen.component';
// import FormBlock from '../../FormBlock/FormBlock.component';

describe('HomeScreen', () => {
  it('should match old snapshot', async () => {
    const tree = renderer.create(<HomeScreen />);
    expect(tree).toMatchSnapshot();
  });
});

describe('getFormBlocks', () => {
  const formBlocks = [
    {
      formName: 'Feedback Form',
      createdAt: '2019-02-27T06:32:52.498Z',
    },
    {
      formName: 'Personal Details Form',
      createdAt: '2019-02-27T06:32:52.501Z',
    },
  ];
  it('should create array of formBlocks element', () => {
    const wrapper = shallow(<HomeScreen />);
    expect(wrapper.instance().getFormBlocks(formBlocks).length).toEqual(2);
  });
  it('should retunr empty array if passed formBlocks array is empty', () => {
    const wrapper = shallow(<HomeScreen />);
    expect(wrapper.instance().getFormBlocks([])).toEqual([]);
  });
});
