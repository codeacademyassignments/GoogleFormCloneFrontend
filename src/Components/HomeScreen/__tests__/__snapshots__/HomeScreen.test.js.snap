// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should match old snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#D6E4F1",
      "flex": 5,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "flex-end",
              "margin": 10,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3F50B8",
                "borderColor": "rgba(0,0,0,0.2)",
                "borderRadius": 100,
                "borderWidth": 1,
                "bottom": 0,
                "height": 70,
                "justifyContent": "center",
                "opacity": 1,
                "right": 45,
                "width": 70,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "height": 20,
                    "width": 5,
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "height": 5,
                    "left": -8,
                    "position": "absolute",
                    "top": 8,
                    "width": 20,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
